app/models/host/managed.rb

```diff
@@ -510,13 +532,18 @@ def attributes_to_import_from_facts
     super + [:domain, :architecture, :operatingsystem]
   end
 
-  def populate_fields_from_facts(facts = self.facts_hash, type = 'puppet')
+  def populate_fields_from_facts(facts = self.facts_hash, type = 'puppet', proxy_id = nil)
     importer = super
     if Setting[:update_environment_from_facts]
       set_non_empty_values importer, [:environment]
     else
       self.environment ||= importer.environment unless importer.environment.blank?
     end
+
+    HostAspects.configuration.registered_aspects.values.each do |aspect_config|
+      aspect_config.model_class.populate_fields_from_facts(self, importer, type, proxy_id)
+    end
+
     operatingsystem.architectures << architecture if operatingsystem && architecture && !operatingsystem.architectures.include?(architecture)
     self.save(:validate => false)
   end
```

Comment:
In puppet implementation proxy_id is needed, as I wrote to @dLobatog, it should be refactored regardless this PR.